ibtk/src/lagrangian/fortran/lagrangian_interaction2d.f.m4

```diff
@@ -2501,36 +2502,34 @@ c
             ic_upper(d) = ic_center(d)+1
          enddo
 
-         ic_lower(0) = max(ic_lower(0),ilower0-nugc0)
-         ic_upper(0) = min(ic_upper(0),iupper0+nugc0)
-
-         ic_lower(1) = max(ic_lower(1),ilower1-nugc1)
-         ic_upper(1) = min(ic_upper(1),iupper1+nugc1)
```

Comment:
From above: since `nugcX` must be nonnegative we can skip all of this and make the loop bounds fixed instead, since `ic_upper(N) - ic_lower(N) == 2`.

Reply:
Notice that `ic_lower` and `ic_upper` give the cell indices of the box that corresponds to the stencil of the regularized kernel function. That box need not be contained in the ghost box (which is determined by `ilowerX-nugcX` and `iupperX+nugcX`). These calls to `min` and `max` are "trimming" the stencil box, to ensure that it is contained in the ghost box.

Two alternatives are:
1. Make sure that the ghost width is large enough that we can ensure that we don't have to do this check. (This may require excluding some IB interaction points that are "far" from the patch interior, and hence cannot contribute to forces in the patch interior.)
2. Move this logic down into the loop where values are actually "spread" --- although putting if statements in the inner loop will probably eliminate the ability of the compiler to generate vectorized code.

Reply:
You're right. I think I know how to fix this. I am surprised that the spread test didn't fail.